api/src/contexts/users/interactors/CreateUserIntreractor.ts

```diff
@@ -0,0 +1,33 @@
+import { IUserRepository } from "../domains/repositories/IUserRepository";
+import { User } from "../domains/entities/User";
+import { hashPassword } from "../../utils/hashPassword";
+import { ICreateUserRequest } from "../usecases/ICreateUserInteractor";
+
+export class CreateUserInteractor {
+  constructor(private readonly userRepository: IUserRepository) {}
+
+  async execute(input: ICreateUserRequest): Promise<User> {
+    const { lastName, firstName, email, password } = input;
+
+    //validation for email
+    const existingUser = await this.userRepository.findByEmail(email);
+    if (existingUser) {
+      throw new Error("Email already exists");
+    }
+
+    // hash password
+    const hashedPassword = await hashPassword(password);
+
+    // exec user
+    const createdUser = await this.userRepository.create({
+      lastName,
+      firstName,
+      email,
+      password: hashedPassword,
+      isResigned: false,
+      role = 1,
```

Comment:
role='USER'だとダメそう？
DB上だとroleの値は 'USER' | 'ADMIN'のはずで、stringをそのまま指定できそうな気がしており
```suggestion
      role = 'USER',
```

Reply:
直しておきました！またStringに代わる関係上User/domains/entities/UserもStringに変えてあります。